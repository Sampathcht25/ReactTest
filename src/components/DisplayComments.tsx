import React, {useState, useEffect} from 'react';
import axios from 'axios';
import '../App.css';

export const DisplayComments = (props: any) => {
    const [getStories, setStories] = useState([]);
    const [getComments, setComments] = useState(null as any)

    const getAPIData = async () =>{
         const result = await axios.get(`https://hacker-news.firebaseio.com/v0/topstories.json?print=pretty`).
                then(({data}) => {
                 return data;
                })
                setStories(result.slice(0, 10));
                return result.slice(0, 10);
    }

    const showCommentsHandler = async(id:number) => {
        const result  = await axios.get(`https://hacker-news.firebaseio.com/v0/item/${id}.json?print=pretty`).then(({data})=>{
        return data;
        });
        setComments(result);
        return result;
    }

    useEffect(()=>{
        getAPIData();
    }, []);


    return (
        <div className="ui container grid">
            <div className="ui row">
                <div className="column eight wide block">
                    <div className="ui devided list">
                        {getStories.map((key, val) => {
                             return(
                                <div className="item" key={`key-${val}`}>
                                    <div className="left floated content" onClick={ ()=> showCommentsHandler(key)}>{key}
                                     </div>
                                </div>
                            )
                        })}
        </div>
        {getComments && 
            <div className="right floated content">{getComments.title}</div>
        }
                </div>
            </div>
        </div>
    )
}