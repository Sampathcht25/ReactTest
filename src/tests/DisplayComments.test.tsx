import React from "react";
import {DisplayComments} from "../components/DisplayComments";
import {shallow} from "enzyme";

describe("DisplayComments Component", () => {
    let wrapper:any;
    beforeEach(()=>{
         wrapper = shallow(<DisplayComments />)
    })
    it("snapshot update", () => {
        expect(wrapper).toMatchSnapshot();
    })
})