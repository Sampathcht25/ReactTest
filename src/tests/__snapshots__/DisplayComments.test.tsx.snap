// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayComments Component snapshot update 1`] = `
<div
  className="ui container grid"
>
  <div
    className="ui row"
  >
    <div
      className="column eight wide block"
    >
      <div
        className="ui devided list"
      />
    </div>
  </div>
</div>
`;
